src/pages/admin/callouts/view/[id]/index.vue

```diff
@@ -134,6 +150,37 @@ async function confirmDeleteCallout() {
   router.push({ path: '/admin/callouts' });
 }
 
+async function endThisCallout() {
+  const now = new Date();
+  const calloutData = {
+    ...props.callout,
+    status: undefined,
+    expires: now,
+  };
+  await updateCallout(props.callout.slug, calloutData);
+  addNotification({
+    title: t('calloutAdmin.ended'),
+    variant: 'success',
+  });
+  router.push({ path: '/admin/callouts' });
+}
+
+async function reopenThisCallout() {
+  const now = new Date();
+  const calloutData = {
```

Comment:
Hmm not sure whether to update the `starts` property or not. I'd be tempted to go for the less "destructive" option here and not overwrite it as they can always update that in the edit interface if they want to.

Also same as above, can probably be simplified to
```ts
await updateCallout(props.callout.slug, {expires: null})
```